### PR TITLE
CI: APIv2 tests: add rootless

### PR DESCRIPTION
**Pull request by edsantiago**
This would've caught a regression that #14549 had to fix.
Let's try to prevent the next regression.

Signed-off-by: Ed Santiago <santiago@redhat.com>

```release-note
None
```

### CODE REVIEW

**Comment by edsantiago** [comment]
@Luap99 two of the failures are in this test which you added this morning in #14549:
https://github.com/containers/podman/blob/874ca426eb51e2b1c9fd1d0c052de2e21f4afd2d/test/apiv2/20-containers.at#L482

[Log](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5521273265586176/html/apiv2-podman-fedora-36-rootless-host.log.html):
```
         not ok 501 [20-containers] GET libpod/containers/sha/json : .HostsPath
         #  expected: 
         #    actual: /run/user/24432/containers/overlay-containers/sha/userdata/hosts
```
Is this a bug in the test, or a bug in the code? I.e., should `.HostsPath` be empty even when rootless?

**Comment by Luap99** [comment]
> Is this a bug in the test, or a bug in the code? I.e., should `.HostsPath` be empty even when rootless?

It must always be empty. It should not depend on root/rootless. I have to double check why it is failing but I definitely tested it rootless on my PR and it worked there.

**Comment by edsantiago** [comment]
Please don't spend any time on it. I added some debug instrumentation which shows that there's a problem restarting the server. I will investigate next week.

File: test/apiv2/test-apiv2 [hunk 1]
```diff
@@ -380,13 +380,26 @@ function start_service() {
...
+    # this will happen as a fork-exec, where the parent podman creates the
+    # namespace and the child is the server. Then, when stop_service() kills
+    # the parent, the child (server) happily stays alive and ruins subsequent
+    # tests that try to restart service with different settings.
+    # Workaround: run an unshare to get namespaces initialized.
+    if [[ $(id -u) != 0 ]]; then
+        # Complicated parens & redirection are to prevent job-control clutter
+        ( $PODMAN_BIN unshare sleep 10 & ) &>/dev/null
+    fi
+
```
Comment (edsantiago) [inline test/apiv2/test-apiv2:394]: @giuseppe @nalind with this addition, CI is now passing. At your convenience, PTAL and see if this is the best (or least-worst) solution, and if the comment is accurate. Thank you for your help in this!

Comment (Luap99) [reply test/apiv2/test-apiv2:394]: I think `podman unshare true` should be enough, the rootless pause process pid is never killed.

Comment (edsantiago) [reply test/apiv2/test-apiv2]: So done, and CI is green. Thanks for the suggestion, that is much cleaner.

**Pull request merged**